File: application/projectone-ui/src/app/examinee/dto/examinee.ts
import { Gender } from "./gender";
export class Examinee {
    public id: string;
    public firstName: string;
    public lastName: string;
    public middleName: string;
    public gender: Gender;
    public dob: Date;
    public examineeId: string;
    public comments: string;
    public version: number;

    constructor(id?: string, firstName?: string, lastName?: string, middleName?: string, gender?: Gender, dob?: Date, examineeId?: string, version?: number) {
        this.id = id;
        this.firstName = firstName;
        this.lastName = lastName;
        this.middleName = middleName;
        this.gender = gender;
        this.dob = dob;
        this.version = version;
    }

    public setId(id: string) {
        this.id = id;
    }

    public getId() {
        return this.id;
    }

    public setFirstName(fname) {
        this.firstName = fname;
    }

    public getFirstName() {
        return this.firstName;
    }

    public setLastName(lname) {
        this.lastName = lname;
    }

    public getLastName() {
        return this.lastName;
    }

    public setMiddleName(middleName) {
        this.middleName = middleName;
    }

    public getMiddleName() {
        return this.middleName;
    }

    public setGender(gender) {
        this.gender = gender;
    }

    public getGender() {
        return this.gender;
    }

    public setDob(dob: Date) {
        this.dob = dob;
    }

    public getDob(): Date {
        return this.dob;
    }

    public getExamineeId() {
        return this.examineeId;
    }

    public setExamineeId(value: string) {
        this.examineeId = value;
    }
    public getVersion() {
        return this.version;
    }

    public setVersion(value: number) {
        this.version = value;
    }

}

File: application/customer-data/src/main/java/com/pearson/projectone/data/entity/customer/examinee/Examinee.java
package com.pearson.projectone.data.entity.customer.examinee;

import com.pearson.projectone.core.support.data.DocumentEntity;
import com.pearson.projectone.data.constants.Gender;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document(collection = "Examinee")
public class Examinee extends DocumentEntity {

	private String examineeId;

	private String hashedIdentifier;

	private String firstName;

	private String lastName;

	private String middleName;

	private Gender gender;

	private Date dob;

	private Date dateOfLastAssessment;

	private String email;

	private String accountId;

	private String comments;

	private boolean archived;

	private boolean practiceMode;

	public String getExamineeId() {
		return examineeId;
	}

	public void setExamineeId(String examineeId) {
		this.examineeId = examineeId;
	}

	public String getFirstName() {
		return firstName;
	}

	public void setFirstName(String firstName) {
		this.firstName = firstName;
	}

	public String getLastName() {
		return lastName;
	}

	public void setLastName(String lastName) {
		this.lastName = lastName;
	}

	public String getHashedIdentifier() {
		return hashedIdentifier;
	}

	public void setHashedIdentifier(String hashedIdentifier) {
		this.hashedIdentifier = hashedIdentifier;
	}

	public String getMiddleName() {
		return middleName;
	}

	public void setMiddleName(String middleName) {
		this.middleName = middleName;
	}

	public Gender getGender() {
		return gender;
	}

	public void setGender(Gender gender) {
		this.gender = gender;
	}

	public Date getDob() {
		return dob;
	}

	public void setDob(Date dob) {
		this.dob = dob;
	}

	public Date getDateOfLastAssessment() {
		return dateOfLastAssessment;
	}

	public void setDateOfLastAssessment(Date dateOfLastAssessment) {
		this.dateOfLastAssessment = dateOfLastAssessment;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getAccountId() {
		return accountId;
	}

	public void setAccountId(String accountId) {
		this.accountId = accountId;
	}

	public String getComments() {
		return comments;
	}

	public void setComments(String comments) {
		this.comments = comments;
	}

	public boolean isArchived() {
		return archived;
	}

	public void setArchived(boolean archived) {
		this.archived = archived;
	}

	public boolean isPracticeMode() {
		return practiceMode;
	}

	public void setPracticeMode(boolean practiceMode) {
		this.practiceMode = practiceMode;
	}
}
